### PR TITLE
simplify import_image

### PR DESCRIPTION
**Pull request by rcerven**
Signed-off-by: Robert Cerven <rcerven@redhat.com>

### CODE REVIEW

File: osbs/api.py [hunk 1]
```diff
@@ -1059,22 +1086,58 @@ def get_image_stream_tag(self, tag_id):
         return self.os.get_image_stream_tag(tag_id)
 
     @osbsapi
-    def ensure_image_stream_tag(self, stream, tag_name, scheduled=False):
+    def ensure_image_stream_tag(self, stream, tag_name, scheduled=False,
+                                reactor_config_dict=None, base_image=False):
```
[the remark this answers is not in the document]
Comment (rcerven) [reply osbs/api.py]: this method is now called just from api.py (_create_build_config_and_build)
we won't use that method from import_image anymore, that is just for fallback

File: osbs/build/build_requestv2.py [hunk 2]
```diff
@@ -101,6 +102,7 @@ def set_params(self, **kwargs):
 
         logger.debug("now setting params '%s' for user_params", kwargs)
         self.user_params.set_params(**kwargs)
+        self.reactor_config_dict = None
```
[the remark this answers is not in the document]
Comment (rcerven) [reply osbs/build/build_requestv2.py]: problem is that we need source registry info in osbs-client in that call from api.py

[the remark this answers is not in the document]
Comment (mlangsdorf) [reply osbs/build/build_requestv2.py]: then retrieve the source registry info and pass that.  You need two fields, so why are you passing around the entire dictionary?

[the remark this answers is not in the document]
Comment (rcerven) [reply osbs/build/build_requestv2.py]: will do

[the remark this answers is not in the document]
Comment (rcerven) [reply osbs/build/build_requestv2.py]: changed, setting only source_registry & organization

File: osbs/build/build_requestv2.py [hunk 3]
```diff
@@ -174,10 +186,12 @@ def set_required_secrets(self):
             data = reactor_config_override
             required_secrets = data.get(req_secrets_key, [])
             token_secrets = data.get(token_secrets_key, [])
+            self._set_data_from_reactor_config(data)
         elif reactor_config_map:
             config_map = self.osbs_api.get_config_map(reactor_config_map)
             required_secrets = config_map.get_data_by_key('config.yaml').get(req_secrets_key, [])
             token_secrets = config_map.get_data_by_key('config.yaml').get(token_secrets_key, [])
+            self._set_data_from_reactor_config(config_map.get_data_by_key('config.yaml'))
```
Comment (mlangsdorf) [inline osbs/build/build_requestv2.py]: maybe refactor this to
```
data = {}
if reactor_config_override:
    data = reactor_config_override
elif reactor_config_map:
    data = self.osbs_api.get_config_map(reactor_config_map).get_data_by_key('config.yaml')
required_secrets = data.get(req_secrets_key, [])
token_secrets = data.get(token_secrets_key, [])
self._set_data_from_reactor_config(data)
```
or better yet, make set_data_from_reactor_config(self) the top level call that figures out data, sets the source stuff, and calls _set_required_secrets(data)

Comment (rcerven) [reply osbs/build/build_requestv2.py]: yeah I'll refactor that

File: osbs/core.py [hunk 4]
```diff
@@ -903,6 +905,70 @@ def import_image(self, name, stream_import, tags=None):
...
+                'from': {"kind": "DockerImage",
+                         "name": '{0}:{1}'.format(repository, tag)},
+                'to': {'name': tag},
+                'importPolicy': {'insecure': insecure},
+                # referencePolicy will default to "type: source"
+                # so we don't have to explicitly set it
+            }
+            stream_import['spec']['images'].append(image_import)
+
+        if not stream_import['spec']['images']:
```
Comment (lcarva) [inline osbs/core.py]: Let's move this to before the for-loop and check `tags_set` instead:
```python
if not tags:
  logger.debug('No tags to import')
  return False

for tag in set(tags):
  ...
```

File: osbs/api.py [hunk 5]
```diff
@@ -1059,22 +1089,55 @@ def get_image_stream_tag(self, tag_id):
...
         img_stream_tag_file = os.path.join(self.os_conf.get_build_json_store(),
                                            'image_stream_tag.json')
         with open(img_stream_tag_file) as f:
             tag_template = json.load(f)
+
+        repository = None
+        registry = None
+        insecure = False
+
+        if (source_registry):
```
Comment (MartinBasti) [inline osbs/api.py]: Extra `(` `)`

File: osbs/api.py [hunk 6]
```diff
@@ -469,9 +470,19 @@ def _create_build_config_and_build(self, build_request):
             existing_bc = self.os.create_build_config(json.dumps(build_json)).json()
 
         if image_stream:
+            source_registry = None
```
[the remark this answers is not in the document]
Comment (rcerven) [reply osbs/api.py]: ok

File: osbs/api.py [hunk 7]
```diff
@@ -988,6 +999,25 @@ def import_image(self, name, tags=None):
             stream_import = json.load(f)
         return self.os.import_image(name, stream_import, tags=tags)
 
+    @osbsapi
+    def import_image_tags(self, name, tags, repository, insecure=False):
+        """Import image tags from specified container registry.
```
[the remark this answers is not in the document]
Comment (rcerven) [reply osbs/api.py]: ok

File: osbs/api.py [hunk 8]
```diff
@@ -1059,22 +1089,55 @@ def get_image_stream_tag(self, tag_id):
...
+                                source_registry=None, organization=None, base_image=None):
         """Ensures the tag is monitored in ImageStream
 
         :param stream: dict, ImageStream object
         :param tag_name: str, name of tag to check, without name of
                               ImageStream as prefix
         :param scheduled: bool, if True, importPolicy.scheduled will be
                                 set to True in ImageStreamTag
+        :param source_registry: dict, source registry from reactor config
+        :param organization: str, oganization for registries
```
[the remark this answers is not in the document]
Comment (rcerven) [reply osbs/api.py]: ok

File: osbs/api.py [hunk 9]
```diff
@@ -1059,22 +1089,55 @@ def get_image_stream_tag(self, tag_id):
...
+    def ensure_image_stream_tag(self, stream, tag_name, scheduled=False,
+                                source_registry=None, organization=None, base_image=None):
         """Ensures the tag is monitored in ImageStream
 
         :param stream: dict, ImageStream object
         :param tag_name: str, name of tag to check, without name of
                               ImageStream as prefix
         :param scheduled: bool, if True, importPolicy.scheduled will be
                                 set to True in ImageStreamTag
+        :param source_registry: dict, source registry from reactor config
```
[the remark this answers is not in the document]
Comment (rcerven) [reply osbs/api.py]: ok

File: osbs/api.py (hunk 5 above)
Comment (rcerven) [reply osbs/api.py]: ok

File: osbs/build/build_requestv2.py [hunk 10]
```diff
@@ -157,28 +160,39 @@ def set_reactor_config(self):
...
 
-    def set_required_secrets(self):
+    def set_data_from_reactor_config(self):
         """
-        Sets required secrets
+        Sets data from reactor config
         """
         reactor_config_override = self.user_params.reactor_config_override.value
         reactor_config_map = self.user_params.reactor_config_map.value
         if not reactor_config_map and not reactor_config_override:
```
[the remark this answers is not in the document]
Comment (rcerven) [reply osbs/build/build_requestv2.py]: ok

File: osbs/core.py (hunk 4 above)
Comment (rcerven) [reply osbs/core.py]: ok

**Comment by rcerven** [comment]
rn updated

**Pull request merged**